pkg/aws/cloud.go

```diff
@@ -42,24 +43,37 @@ type Cloud interface {
 
 // NewCloud constructs new Cloud implementation.
 func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud, error) {
-	metadataSess := session.Must(session.NewSession(aws.NewConfig()))
-	metadata := services.NewEC2Metadata(metadataSess)
-	if len(cfg.Region) == 0 {
```

Comment:
why not modify this block instead?

Reply:
I actually added the IMDS call inside the if to reduce the number of IMDS calls per instance. So I am making the call only if it is required.